src/eventProcessor/GoogleAnalyticsEventProcessor.js

```diff
@@ -0,0 +1,129 @@
+goog.module('measurementLibrary.eventProcessor.GoogleAnalyticsEventProcessor');
+
+/**
+ * A class that processes events pushed to the data layer
+ * by constructing and sending Google Analytics events via
+ * Measurement Protocol.
+ * For its Google Analytic events, it generates a unique client ID
+ * and stores it in the long term storage model provided.
+ * @implements {EventProcessor}
+ */
+class GoogleAnalyticsEventProcessor {
+  /**
+   * @param {{
+   *     'api_secret': (string|undefined),
+   *     'measurement_id': (string|undefined),
+   *     'measurement_url': string,
+   *     'client_id_expires': number,
+   *     'automatic_params': !Object<string, boolean>,
+   * }=} optionsObject
+   */
+  constructor({
+      'api_secret': apiSecret,
+      'measurement_id': measurementId,
+      'measurement_url': measurementUrl = 'https://www.google-analytics.com/mp/collect',
+      'client_id_expires': clientIdExpires = Number.POSITIVE_INFINITY,
+      'automatic_params': userAutomaticParams = {},
+    } = {}) {
+    /**
+     * Parameters that are important to all events and will be searched for
+     * globally in the data model.
+     * @private @const {!Object<string, boolean>}
+     */
+    this.automaticParams_ = Object.assign({
+      'page_path': true,
+      'page_location': true,
+      'page_title': true,
+      'user_id': true,
+      'client_id': true,
+    }, userAutomaticParams);
```

Comment:
This is just a set, right? Eww. (Though I can't think of a better way to do it, maybe leave a comment? Actually thee guidelines say Object.assign adds a polyfill, you can check it https://docs.google.com/document/d/1XIhihp4elYAa9cFqJL9vg3f7u4kbsoZqIQa5ytBqYg0/edit#heading=h.3xoqh97uc7cf

Reply:
Per my other comment, if the user params were an array, this could be addressed by changing the code here to:

```
this.automaticParams_ = { /* default auto params */ };
for (let i = 0; i < userAutomaticParams.length; i++) {
  this.automaticParams_[userAutomaticParams[i]] = true;
}
```

Reply:
So this is one of the only ways represent a set in JS pre-ES6 without defining a custom data structure. I added more documentation surrounding this parameter that makes it explicitly clear what a user must do to override a default value / add their own.